src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -387,7 +387,28 @@ public void Dispose()
                 return;
             }
 
-            // todo: log unprocessed scheduler registrations?
+            // Execute all outstanding work items
+            foreach (var task in stopped.Result)
+            {
+                try
+                {
+                    task.Action.Run();
```

Comment:
So this was the major fix - the JVM has equivalent code too; need to make sure everything gets processed upon `IScheduler` shutdown.